curvlinops/inverse.py

```diff
@@ -327,30 +337,69 @@ def _compute_inverse_factors(
             return (aaT_eigvecs, aaT_eigvals), (ggT_eigvecs, ggT_eigvals)
         else:
             damping_aaT, damping_ggT = self._compute_damping(aaT, ggT)
-            aaT_inv = (
-                None
-                if aaT is None
-                else cholesky_inverse(
-                    cholesky(
+
+            # Compute inverse of aaT via Cholesky decomposition
+            try:
+                aaT_chol = (
+                    None
+                    if aaT is None
+                    else cholesky(
                         aaT.add(
                             eye(aaT.shape[0], dtype=aaT.dtype, device=aaT.device),
                             alpha=damping_aaT,
                         )
                     )
                 )
-            )
-            ggT_inv = (
-                None
-                if ggT is None
-                else cholesky_inverse(
-                    cholesky(
+            except Exception as exception:
```

Comment:
Should we specify the exception type further to make sure the re-try happens only if cholesky crashes due non-convergence?

Reply:
Let's go for `RuntimeError`